src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -0,0 +1,884 @@
+
+  const [applicationCardsOpen, setApplicationCardsOpen] = useState(false);
+  const [newEditorProfile, setNewEditorProfile] = useState(null); // Stores the StudentProfileInfo of the new editor before the user confirms the change
+  const [changeEditorDialogOpen, setChangeEditorDialogOpen] = useState(false);
+  const [submitDialogOpen, setSubmitDialogOpen] = useState(false); // Use this for submitting app (later feature)
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarText, setSnackbarText] = useState('');
+  const [snackbarSeverity, setSnackbarSeverity] = useState('');
+  const [saveButtonAlertTimeout, setSaveButtonAlertTimeout] = useState(null);
+
+  /**
+   * Loads the user's saved apartment application, if one exists
+   */
+  const loadSavedApplication = useCallback(async () => {
+    // TODO: Implement this once save/load of application data has been implemented in the backend
+    setLoading(true);
+    // Check if the current user is on an application. Returns the application ID number if found
+    let newApplicationID = null;
+    try {
+      newApplicationID = await housing.getCurrentApplicationID();
+      //! DEBUG
+      console.log('Retrieved Application ID: ' + newApplicationID);
+      if (newApplicationID === null || newApplicationID === -1) {
+        // Intentionally trigger the 'catch'
+        throw new Error("Invalid value of 'newApplicationID' = " + newApplicationID);
+      }
+      setApplicationID(newApplicationID);
+      let applicationDetails = await housing.getApartmentApplication(newApplicationID);
+      if (applicationDetails) {
+        if (applicationDetails.DateSubmitted) {
+          setDateSubmitted(applicationDetails.DateSubmitted);
+        }
+        if (applicationDetails.DateModified) {
+          setDateModified(applicationDetails.DateModified);
+        }
+        if (applicationDetails.Username) {
+          setEditorUsername(applicationDetails.Username);
+        }
+        if (applicationDetails.Applicants) {
+          applicationDetails.Applicants.forEach(async (applicantInfo) => {
+            const newApplicantProfile = await user.getProfileInfo(applicantInfo.Username);
+            setApplicants((prevApplicants) =>
+              prevApplicants.concat({
+                Profile: newApplicantProfile,
+                OffCampusProgram: applicantInfo.OffCampusProgram,
+              }),
+            );
+          });
+        }
+      }
+    } catch {
+      // No existing application was found in the database,
+      // or an error occurred while attempting to load the application
+      setApplicationID(-1);
+      if (!editorUsername) {
+        if (
+          applicants.every((applicant) => applicant.Profile.AD_Username !== userProfile.AD_Username)
+        ) {
+          setApplicants((prevApplicants) =>
+            prevApplicants.concat({ Profile: userProfile, OffCampusProgram: '' }),
```

Comment:
I'm not sure `Array.concat()` is the best method to use here. That method is intended to join two arrays together. If you're simply appending an element to the array, you could use `Array.push()`, or better yet just the spread operator, e.g.:
```suggestion
            [...prevApplicants, {Profile: userProfile, OffCampusProgram: '' }]
```

Reply:
Interesting, I like it.
I have only recently started to learn more about how and when you can use the spread operator